src/index.ts

```diff
@@ -1,66 +1,54 @@
-'use strict'
-
-const assert = require('debug')
-const debug = require('debug')
-const log = debug('libp2p:webrtcdirect')
-log.error = debug('libp2p:webrtcdirect:error')
-const errcode = require('err-code')
-
-const wrtc = require('wrtc')
-const SimplePeer = require('libp2p-webrtc-peer')
-const isNode = require('detect-node')
-const mafmt = require('mafmt')
-const { base58btc } = require('multiformats/bases/base58')
-const { fetch } = require('native-fetch')
-const withIs = require('class-is')
-const { AbortError } = require('abortable-iterator')
-const { toString } = require('uint8arrays/to-string')
-const { fromString } = require('uint8arrays/from-string')
-
-const { CODE_CIRCUIT, CODE_P2P } = require('./constants')
-const toConnection = require('./socket-to-conn')
-const createListener = require('./listener')
-
-function noop () {}
-
-/**
- * @typedef {import('multiaddr').Multiaddr} Multiaddr
- */
-
-class WebRTCDirect {
-  /**
-   * @class
-   * @param {object} options
-   * @param {Upgrader} options.upgrader
-   */
-  constructor ({ upgrader }) {
-    assert(upgrader, 'An upgrader must be provided. See https://github.com/libp2p/interface-transport#upgrader.')
+import { logger } from '@libp2p/logger'
+import errcode from 'err-code'
+// @ts-expect-error no types
+import wrtc from 'wrtc'
+// @ts-expect-error no types
+import SimplePeer from 'libp2p-webrtc-peer'
+// @ts-expect-error no types
+import isNode from 'detect-node'
+import mafmt from 'mafmt'
+import { base58btc } from 'multiformats/bases/base58'
+import { fetch } from 'native-fetch'
+import { AbortError } from 'abortable-iterator'
+import { toString } from 'uint8arrays/to-string'
+import { fromString } from 'uint8arrays/from-string'
+import { CODE_CIRCUIT, CODE_P2P } from './constants'
+import { toMultiaddrConnection } from './socket-to-conn'
+import { createListener } from './listener'
+
+import type { AbortOptions } from '@libp2p/interfaces'
+import type { Listener, ListenerOptions, Transport, Upgrader } from '@libp2p/interfaces/transport'
+import type { Multiaddr } from '@multiformats/multiaddr'
+
+const log = logger('libp2p:webrtcdirect')
+
+export interface WebRTCDirectListenerOptions extends ListenerOptions{
+  channelOptions?: Object
```

Comment:
I followed the same logic as in [js-libp2p-webrtc-star/webrtc-star-transport](https://github.com/libp2p/js-libp2p-webrtc-star/blob/master/packages/webrtc-star-transport/src/index.ts#L65). However I didn't describe the type of `channelOptions`. As far as this code is concerned `Object` type looked fine but let me know if you want to describe it thoroughly.

Reply:
Please look. The exact shape is described here https://www.npmjs.com/package/libp2p-webrtc-peer#api. See constructor options:
```
{
  initiator: false,
  channelConfig: {},
  channelName: '<random string>',
  config: { iceServers: [{ urls: 'stun:stun.l.google.com:19302' }, { urls: 'stun:global.stun.twilio.com:3478?transport=udp' }] },
  offerOptions: {},
  answerOptions: {},
  sdpTransform: function (sdp) { return sdp },
  stream: false,
  streams: [],
  trickle: true,
  allowHalfTrickle: false,
  wrtc: {}, // RTCPeerConnection/RTCSessionDescription/RTCIceCandidate
  objectMode: false
}
```
I described all the keys of this object. However for `config` key I used a plain object. Do you want me to describe it more rigid with `iceServers` and `urls`?